sdk/src/logs/logger_provider.cc

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "opentelemetry/sdk/logs/logger_provider.h"
+
+OPENTELEMETRY_BEGIN_NAMESPACE
+namespace sdk
+{
+namespace logs
+{
+LoggerProvider::LoggerProvider(std::shared_ptr<LogProcessor> processor) noexcept
+    : processor_{processor}
+{}
+
+LoggerProvider::LoggerProvider() noexcept : processor_{nullptr} {}
+
+opentelemetry::nostd::shared_ptr<opentelemetry::logs::Logger> LoggerProvider::GetLogger(
+    opentelemetry::nostd::string_view name,
+    opentelemetry::nostd::string_view options) noexcept
+{
+  // If a logger with a name "name" already exists, return it
+  std::unordered_map<std::string, std::shared_ptr<Logger>>::iterator loggerkv =
+      loggers_.find(name.data());
+  if (loggerkv != loggers_.end())
+  {
+    return opentelemetry::nostd::shared_ptr<Logger>(loggerkv->second);
+  }
+
+  // If no logger with that name exists yet,
+  // Create it and add it to the map of loggers
+  auto loggerInstance = new Logger(GetProcessor());
+  std::shared_ptr<Logger> loggerPtr{loggerInstance};
+  loggers_[name.data()] = loggerPtr;
```

Comment:
Is this thread safe?

Reply:
Right... if two threads simultaneously call GetLogger(name) then the map can be written to while the other thread is reading/writing from it. I'll fix this by using a spin lock, like what we did for the [API's provider](https://github.com/open-o11y/opentelemetry-cpp/blob/a2d883471ba422d7a3ddfcb435e794eb7958a5f3/api/include/opentelemetry/logs/provider.h#L52).

Reply:
Is it already specified what the exact behavior of this method should be, in relation to the passed logger name?

If not, maybe for now you can just do what the `TracerProvider` does and return a new `Logger`, regardless of the name?

With the current solution, I'm not sure about the benefit how having differently named loggers that all share the very same processor.

Reply:
Yea right now having all the loggers reference the same processor is pretty much the same as having a single logger instance (like in traces). The logging spec isn't well defined right now so we took a hybrid approach of basing our implementation off traces while also trying to offer functionality that common logging libraries have, such as creating unique logger instance. When the logging spec gets more well defined we can change our implementation to match with it, for example having a unique processor for each logger.